### ISSUE TITLE
Cannot establish connection on edge browser

### ISSUE THREAD

**Issue by betimer** [socketio/socket.io-client#1203]
*Note*: for support questions, please use one of these channels: [stackoverflow](http://stackoverflow.com/questions/tagged/socket.io) or [slack](https://socketio.slack.com)

### You want to:

* [x] report a *bug*
* [ ] request a *feature*

### Current behaviour

Cannot connect to socket.io from edge browser.
It looks like the client keeps asking initial connection request.

### Setup
- OS: Win 10 x64
- browser: Edge
- socket.io version: 2.1 (both client and server)

### Other information (e.g. stacktraces, related issues, suggestions how to fix)

![image](https://user-images.githubusercontent.com/8111748/39501532-d344bd1c-4dfe-11e8-92d7-7e94bd8e3760.png)

**Comment by darrachequesne** [comment]
Are you able to connect to https://socket.io/demos/chat/ ?

The value `(from cache)` in the column `Received` seems surprising though.

**Comment by betimer** [comment]
@darrachequesne yes that page is working on edge, but just my socket-io-client does not work.
May I know which socket io client version on  https://socket.io/demos/chat/ is using?

Maybe i can try same version for my code.
Thanks.
